plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/startup/AmazonQStartupActivity.kt

```diff
@@ -3,13 +3,25 @@
 
 package software.aws.toolkits.jetbrains.services.amazonq.startup
 
+import com.intellij.openapi.application.runInEdt
 import com.intellij.openapi.project.Project
 import com.intellij.openapi.startup.ProjectActivity
+import com.intellij.openapi.wm.ToolWindowManager
 import software.aws.toolkits.jetbrains.services.amazonq.toolwindow.AmazonQToolWindow
+import software.aws.toolkits.jetbrains.services.amazonq.toolwindow.AmazonQToolWindowFactory
+import software.aws.toolkits.jetbrains.services.codewhisperer.explorer.CodeWhispererExplorerActionManager
 
 class AmazonQStartupActivity : ProjectActivity {
     override suspend fun execute(project: Project) {
         // initialize html contents in BGT so users don't have to wait when they open the tool window
         AmazonQToolWindow.getInstance(project)
+
+        if (CodeWhispererExplorerActionManager.getInstance().getIsFirstRestartAfterQInstall()) {
```

Comment:
for 241 failure, maybe move the application service declaration to `module-amazonq.xml` from `plugin-codewhisperer.xml`

Reply:
it works